pkg/node/compute.go

```diff
@@ -269,77 +293,70 @@ func NewComputeNode(
-				RejectStatelessJobs: config.JobSelectionPolicy.RejectStatelessJobs,
-			}),
-			semantic.NewProviderInstalledStrategy(
-				publishers,
-				func(j *models.Job) string { return j.Task().Publisher.Type },
-			),
-			semantic.NewStorageInstalledBidStrategy(storages),
-			semantic.NewInputLocalityStrategy(semantic.InputLocalityStrategyParams{
-				Locality: config.JobSelectionPolicy.Locality,
-				Storages: storages,
-			}),
-			semantic.NewExternalCommandStrategy(semantic.ExternalCommandStrategyParams{
-				Command: config.JobSelectionPolicy.ProbeExec,
-			}),
-			semantic.NewExternalHTTPStrategy(semantic.ExternalHTTPStrategyParams{
-				URL: config.JobSelectionPolicy.ProbeHTTP,
-			}),
-			executor_util.NewExecutorSpecificBidStrategy(executors),
-		}
-	} else {
-		semanticBidStrats = []bidstrategy.SemanticBidStrategy{config.BidSemanticStrategy}
+	semanticBidStrats := []bidstrategy.SemanticBidStrategy{
+		semantic.NewNetworkingStrategy(cfg.BacalhauConfig.JobAdmissionControl.AcceptNetworkedJobs),
+		semantic.NewStatelessJobStrategy(semantic.StatelessJobStrategyParams{
+			RejectStatelessJobs: cfg.BacalhauConfig.JobAdmissionControl.RejectStatelessJobs,
+		}),
+		semantic.NewProviderInstalledStrategy(
+			publishers,
+			func(j *models.Job) string { return j.Task().Publisher.Type },
+		),
+		semantic.NewStorageInstalledBidStrategy(storages),
+		semantic.NewInputLocalityStrategy(semantic.InputLocalityStrategyParams{
+			Locality: cfg.BacalhauConfig.JobAdmissionControl.Locality,
+			Storages: storages,
+		}),
+		semantic.NewExternalCommandStrategy(semantic.ExternalCommandStrategyParams{
+			Command: cfg.BacalhauConfig.JobAdmissionControl.ProbeExec,
+		}),
+		semantic.NewExternalHTTPStrategy(semantic.ExternalHTTPStrategyParams{
+			URL: cfg.BacalhauConfig.JobAdmissionControl.ProbeHTTP,
+		}),
+		executor_util.NewExecutorSpecificBidStrategy(executors),
 	}
 
-	var resourceBidStrats []bidstrategy.ResourceBidStrategy
-	if config.BidResourceStrategy == nil {
-		resourceBidStrats = []bidstrategy.ResourceBidStrategy{
-			resource.NewMaxCapacityStrategy(resource.MaxCapacityStrategyParams{
-				MaxJobRequirements: config.JobResourceLimits,
-			}),
-			executor_util.NewExecutorSpecificBidStrategy(executors),
-		}
-	} else {
-		resourceBidStrats = []bidstrategy.ResourceBidStrategy{config.BidResourceStrategy}
+	resourceBidStrats := []bidstrategy.ResourceBidStrategy{
+		resource.NewMaxCapacityStrategy(resource.MaxCapacityStrategyParams{
+			MaxJobRequirements: allocatedResources,
+		}),
+		executor_util.NewExecutorSpecificBidStrategy(executors),
 	}
 
 	return compute.NewBidder(compute.BidderParams{
-		NodeID:           nodeID,
+		NodeID:           cfg.NodeID,
 		SemanticStrategy: semanticBidStrats,
 		ResourceStrategy: resourceBidStrats,
 		Store:            executionStore,
 		Callback:         computeCallback,
 		Executor:         bufferRunner,
-		GetApproveURL: func() *url.URL {
-			return apiServer.GetURI().JoinPath("/api/v1/compute/approve")
-		},
```

Comment:
If we are removing this, is there any point including `NewExternalHTTPStrategy`?

Reply:
no that is still functional. This approval uri was introduced with insulated jobs